### PR TITLE
The brand new Live Release or Work Pane

### PR DESCRIPTION
**Pull request by nid90**
Redo how a release progresses in Tramline, both visually and logically.

Tramline Test 1 _Owner_ _ Tramline

This is a giant PR, that essentially rewrites the core functionality of Tramline:

- Remove the step/step-run/deployment/deployment-run abstraction completely
- Rally the release around a first-class idea of a `Build`
- Smaller, sub-components are now first-class, like WorkflowRun
- Wrap Production releases in a special (and first-class) way, ditto for Pre-Prod releases
- Redo how configuration works for a release. Configuration is now safe to edit at any point in time.
- Visually lay out the "live release" or the "work pane" in a vertical tab system with status information
- Each tab represents a meaningful "work area" and multiple tabs are grouped based on broad categories
- New changes are clearly and loudly notified to the user in different parts of the live release pane
- Builds can now be switched and selected during submissions (not always the latest)
- Release notes can be edited for multiple locales

There's [a lot more to do here](https://github.com/orgs/tramlinehq/projects/12), but this is a releasable alpha cut that we can try with a few users.

## Developer Notes

List of feature flags removed:
1. team_analysis_enabled
2. reldex_enabled
3. release_monitoring
4. allow_blocked_step
5. fixed_build_number
6. custom_release_version

### CODE REVIEW

File: app/components/notification_settings_component.rb [hunk 1]
```diff
@@ -25,15 +25,35 @@ class NotificationSettingsComponent < ViewComponent::Base
...
+    workflow_run_unavailable: {icon: "v2/alert_circle.svg", description: "A workflow run was not found"}
   }.map
     .with_index { |(key, value), index| [key, value.merge(number: index.succ)] }
     .to_h
     .with_indifferent_access
 
-  unless Set.new(NOTIFICATIONS.keys).eql?(Set.new(NotificationSetting.kinds.keys))
-    raise InvalidNotificationSettings
-  end
+  # unless Set.new(NOTIFICATIONS.keys).eql?(Set.new(NotificationSetting.kinds.keys))
```
Comment (kitallis) [inline app/components/notification_settings_component.rb:54]: Should this be uncommented?

File: app/controllers/beta_releases_controller.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  include Tabbable
+  before_action :require_write_access!
+  before_action :set_release_platform_run, only: %i[create]
+
+  def index
+    live_release!
+    @app = @release.app
+  end
+
+  def create
```
Comment (nid90) [inline app/controllers/beta_releases_controller.rb:11]: ensure the release platform run is active. either here or in Coordinator action.

File: app/controllers/internal_releases_controller.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class InternalReleasesController < SignedInApplicationController
+  include Tabbable
+  before_action :require_write_access!
```
Comment (nid90) [inline app/controllers/internal_releases_controller.rb:3]: why?

File: app/controllers/pre_prod_releases_controller.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
+class PreProdReleasesController < SignedInApplicationController
+  before_action :require_write_access!
```
Comment (nid90) [inline app/controllers/pre_prod_releases_controller.rb:2]: no

File: app/controllers/production_releases_controller.rb [hunk 5]
```diff
@@ -0,0 +1,37 @@
+class ProductionReleasesController < SignedInApplicationController
+  before_action :require_write_access!
```
Comment (nid90) [inline app/controllers/production_releases_controller.rb:2]: conditionally

File: app/controllers/production_releases_controller.rb [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  before_action :set_release_platform_run, only: %i[create]
+  before_action :set_prod_release, only: %i[changes_since_previous]
+  before_action :set_app, only: %i[changes_since_previous]
+
+  def create
```
Comment (nid90) [inline app/controllers/production_releases_controller.rb:7]: ensure the release platform run is active. either here or in Coordinator action.

File: app/controllers/store_rollouts_controller.rb [hunk 7]
```diff
@@ -0,0 +1,86 @@
+class StoreRolloutsController < SignedInApplicationController
+  include Tabbable
+
+  before_action :require_write_access!
```
Comment (nid90) [inline app/controllers/store_rollouts_controller.rb:4]: conditionally. all except index.

File: app/controllers/store_submissions_controller.rb [hunk 8]
```diff
@@ -1,101 +1,119 @@
 class StoreSubmissionsController < SignedInApplicationController
+  include Mocks::Sandboxable
+  include Tabbable
+
   before_action :require_write_access!
```
Comment (nid90) [inline app/controllers/store_submissions_controller.rb:5]: conditionally. all except index.

File: app/controllers/workflow_runs_controller.rb [hunk 9]
```diff
@@ -0,0 +1,29 @@
+class WorkflowRunsController < SignedInApplicationController
+  before_action :set_workflow_run, only: [:trigger, :retry]
```
Comment (nid90) [inline app/controllers/workflow_runs_controller.rb:2]: require write access

File: app/components/v2/release_list_component.html.erb [hunk 10]
```diff
@@ -44,6 +44,10 @@
       <% modal.with_button(label: "Prepare new release", scheme: :default, type: :action, size: :xxs, disabled: !release_startable?)
               .with_icon("play.svg", size: :md) %>
       <% modal.with_body do %>
+        <div class="mb-6">
+          <%= render V2::EmptyStateComponent.new(text: branch_help, banner_image: "v2/git_branch.svg", type: :tiny) %>
```
Comment (nid90) [inline app/components/v2/release_list_component.html.erb:48]: for later maybe: don't use empty state component

**Pull request merged**